packages/experiments/src/components/Button/Button.view.tsx

```diff
@@ -1,19 +1,23 @@
 /** @jsx withSlots */
 import { Stack, Text, KeytipData } from 'office-ui-fabric-react';
 import { withSlots, getSlots } from '../../Foundation';
-import { getNativeProps, buttonProperties } from '../../Utilities';
+import { getNativeProps, anchorProperties, buttonProperties } from '../../Utilities';
 import { Icon } from '../../utilities/factoryComponents';
 
 import { IButtonComponent, IButtonProps, IButtonRootElements, IButtonSlots, IButtonViewProps } from './Button.types';
 
 export const ButtonView: IButtonComponent['view'] = props => {
   const { icon, content, children, disabled, onClick, allowDisabledFocus, ariaLabel, keytipProps, buttonRef, ...rest } = props;
 
+  const rootType = _deriveRootType(props);
+  const htmlType = rootType === 'a' ? 'link' : 'button';
+  const propertiesType = rootType === 'a' ? anchorProperties : buttonProperties;
```

Comment:
This is starting to make me feel that this view is doing too much and button vs anchor should be split out into smaller components. That way you can check once if there is an href and then have each component handle it's own props correctly.

Reply:
Or maybe there should be a function like _deriveRootState or something that bundles all that work into one place.

Reply:
I was thinking about the second option you mentioned yesterday and I think I'm going to go ahead and combine those 3 lines into one function.

Reply:
@joschect Updated with the function approach bundling that work 👍